### PR TITLE
Feature to override http client

### PR DESCRIPTION
**Pull request by dkropachev**
This feature could be used for multiple purposes:
from custom retry to ssl CA injection

### CODE REVIEW

**Review by scylladb-buff — COMMENTED** [review]
LGTM

File: go/v2/example_test.go [hunk 1]
```diff
@@ -22,6 +26,28 @@ func ExampleAlternatorNodes() {
...
+		}
+
+		if !systemPool.AppendCertsFromPEM(customPEMCertificate) {
+			panic("failed to append custom certificate")
+		}
+
+		alternatorNodes.SetHTTPClient(&http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
```
Comment (ksinica) [inline go/v2/example_test.go:44]: This is definitely not critical, but I'm not sure if we want to suggest skipping verification of the certificate chain to clients.

File: go/v2/alternator_lb.go [hunk 2]
```diff
@@ -35,12 +40,18 @@ type AlternatorNodes struct {
...
-		nodes:  initialNodes,
+		scheme:     scheme,
+		port:       port,
+		nodes:      initialNodes,
+		httpClient: http.DefaultClient,
 	}
 }
 
+// SetHTTPClient changes underlying http client used for fetching the list of alternator nodes.
+func (n *AlternatorNodes) SetHTTPClient(client httpClient) {
```
Comment (ksinica) [inline go/v2/alternator_lb.go:51]: Cannot we just use `*http.Client` argument here? A distinct type, rather a private interface, would communicate more clearly what this function does. `http.Client` is also easily mockable via the `http.RoundTripper` interface provided in the Transport field, so we are not gaining that much from asserting a "doer" IMO.

Comment (dkropachev) [reply go/v2/alternator_lb.go:51]: Interface provide you more flexibility in that regard, if you look at `http.Client` code, so many things done before it reaches `rt.RoundTrip(req)`.

File: go/v2/example_test.go (hunk 1 above)
Comment (dkropachev) [reply go/v2/example_test.go]: removed

**Comment by dkropachev** [comment]
@sylwiaszunejko , can you please review it.

**Comment by nyh** [comment]
> This feature could be used for multiple purposes: from custom retry to ssl CA injection

Please in commit messages the language and if relevant driver version (e.g, Java SDK v1 vs v2) that this commit pertains to. This patch, for example, is for the Go language.

**Comment by nyh** [comment]
By the way, @dkropachev I see you are writing many "dk/*" branches to this repository. Please consider adding to this public repository only branches that you consider to be useful for everyone (e.g., releases, etc.) - you'd better put your private branches in your own fork of the repository and not push all the branches into the main repsitory. This will make the branch list cleaner for everyone.

**Pull request closed without merging**